README.md

```diff
@@ -1,2 +1,319 @@
+which takes in a message and some configurations and creates and shows a toast in the DOM.
+This is more convenient for one-off toasts,
+or for JavaScript-driven situations,
+similar to how `alert()` is currently used.
+
+```js
+import { showToast } from 'std:elements/toast';
+
+const toast = showToast("Hello World!", {
+    theme: "info",
+    duration: 3000
+});
+```
+
+## Goals
+
+Across popular toast implementations there are recurring patterns,
+which the standard toast aims to accomplish natively.
+
+- The component will be accessible by default;
+  native accessibility is is a strong priority for toasts,
+  as they can be difficult to make properly accessible.
+
+- Toast implementations are often shaped similarly,
+  and a goal of the standard toast is to make it as easy as possible for developers
+  to build and style toasts that conform to those common shapes.
+
+    - Toasts almost always support a message, action, and close button.
+
+    - Toasts often support an icon and a title.
+
+- The positioning of the toast must be intuitive,
+  so the standard toast will come with built-in support for common positions,
+  as well as a sensible default.
+
+- To balance ease of use with customization,
+  the standard toast will support creating and showing with one JavaScript function,
+  as well as writing a custom view with a `<std-toast>` element
+  and showing that with a method.
+
+- The standard toast will come with support for showing multiple toasts,
+  either by stacking them in the view,
+  or queueing them and displaying sequentially.
+
+## Proposed API
+
+The element is provided as a [built-in module](https://github.com/tc39/proposal-javascript-standard-library/blob/master/README.md),
+named `"std:elements/toast"`.
+
+### The `<std-toast>` element
+
+#### Attributes
+
+- [Global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)
+- `open`: a boolean attribute, determining whether the toast is visible or not (according to the default styles). By default toasts are not shown.
+- `theme`: one of `"default"`, ???, or ???, conveying the semantic priority of the toast, and influencing its styling (both default and user-provided)
+    - TODO: decide on list of themes to natively support and create and style them.
+- `position`: one of `"top-left"`, `"top-center"`, `"top-right"`, `"middle-left"`, `"middle-center"`, `"middle-right"`, `"bottom-left"`, `"bottom-center"`, or `"bottom-right"`. The default (if the attribute is omitted or set to an invalid value) is `"bottom-right"`.
+    - TODO: this seems stylistic; should it be controlled via CSS instead?
+    - TODO: these are physical positions; should we use [logical ones](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties/Basic_concepts) instead?
+- `closebutton`: a boolean attribute, determining whether an explicit close button is shown. By default toasts do not have a close button. TODO
```

Comment:
Where would such a button appear? In particular, is there an area of the toast that is somehow reserved for the system. For the toast to be usable as the foundation of other library's toasts, we need to allow total control (including drawing outside of the box) but we also need make it work well when the close button is requested. Maybe it just means that a toast without a native close button draws in the full area and one with one draws in an area that makes sense given the positioning of the button.

BTW, me asking this question is what prompted me to send the mail about considering moving this to a doc.